src/meltano/core/tracking/tracker.py

```diff
@@ -3,10 +3,13 @@
 from __future__ import annotations
 
 import datetime
+import json
 import locale
 import re
+import uuid
 from contextlib import contextmanager
-from typing import Any
+from pathlib import Path
+from typing import Any, NamedTuple, Optional, Tuple, Union
```

Comment:
@aaronsteers  that from __future__ import annotations allows new style annotations, and we've been slowly moving the code base to those. That allows referring list , tuple, and friends without having to import anything from the typing package.

If you're running the pre-commit hook pyupgrade will catch a lot of these for you, but figured I'd give a ya heads up.

Reply:
Updated per your suggestion. Thanks!

I considered putting them in the new style before, but my IDE linter is complaining about the new types. Going forward, I may need to find a fix for the mypy lint issue, perhaps using a higher Python version for my development, since it seems mypy prefers 3.9 or 3.10.